examples/plot_partial_dependence_visualization_api.py

```diff
@@ -0,0 +1,123 @@
+"""
+=========================================
+Advanced Plotting With Partial Dependence
+=========================================
+The :func:`~sklearn.inspection.plot_partial_dependence` function returns a
+:class:`~sklearn.inspection.PartialDependenceDisplay` object that can be used
+for plotting without needing to recalculate the partial dependence. In this
+example, we show how to plot partial dependence plots and how to quickly
+customize the plot with the Visualization API.
+
+.. note::
+
+    See also :ref:`sphx_glr_auto_examples_plot_roc_curve_visualization_api.py`
+
+"""
+print(__doc__)
+
+import matplotlib.pyplot as plt
+from sklearn.datasets import load_boston
+from sklearn.neural_network import MLPRegressor
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import StandardScaler
+from sklearn.pipeline import make_pipeline
+from sklearn.tree import DecisionTreeRegressor
+from sklearn.inspection import plot_partial_dependence
+
+
+##############################################################################
+# Train models on the boston housing price dataset
+# ================================================
+#
+# First, we load the boston housing price dataset and split the the dataset
+# into a training and test set. Then, we train a decision tree and a
+# multi-layer perceptron on the training set.
+
+boston = load_boston()
+X, y = boston.data, boston.target
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1,
```

Comment:
What is the rationale for using a test set?

Note that for GBDT estimators the partial dependence is always computed based on the training data with the recursion method.